File: Data Structures & Algorithms/subsets/submission-0.java
class Solution {
    void backtrack(int[] nums, int idx, List<List<Integer>> result, List<Integer> path) {
        result.add(new ArrayList<>(path));

        for (int i = idx; i < nums.length; i++) {
            path.add(nums[i]);
            backtrack(nums, i + 1, result, path);
            path.remove(path.size() - 1);
        }
    }

    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> result = new ArrayList<>();
        backtrack(nums, 0, result, new ArrayList<>());
        return result;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.go
import "slices"

func backtrack(result *[][]int, nums []int , idx int, path []int, sum int){
    if sum == 0 {
       
        pathCopy := make([]int, len(path))
		copy(pathCopy, path)

        *result = append(*result,pathCopy)
        return
    }

    if idx>=len(nums){
        return
    }

    for i:=idx ; i<len(nums); i++{
        if sum < nums[i]{
            break;
        }
        path = append(path,nums[i])
        backtrack(result,nums,i,path,sum-nums[i])
        path = path[:len(path)-1]
    }
}



func combinationSum(nums []int, target int) [][]int {
    var result [][]int
    var path []int

    slices.Sort(nums)

    backtrack(&result,nums,0,path,target)
    return result
}
